src/platform_linux/shmem.c

```diff
@@ -76,6 +108,65 @@ typedef struct shm_frag_info {
  */
 #define SHM_NUM_LARGE_FRAGS (MAX_THREADS * 2)
 
+/*
+ * Currently, we track free-fragments in lists limited to these sizes.
+ * Each such free-list has a head-list field in shared memory control block.
+ * of sizes.This list can be easily expanded.
+ */
+#define SHM_SMALL_FRAG_MIN_SIZE 64  // Will not track for size < min bytes
+#define SHM_SMALL_FRAG_MAX_SIZE 512 // Will not track for size > max bytes
+
+/*
+ * Each free fragment that is hooked into the free-list is described by this
+ * tiny tracking structure.
+ */
+typedef struct free_frag_hdr {
+   struct free_frag_hdr *free_frag_next;
+   size_t                free_frag_size;
+} free_frag_hdr;
+
+/*
+ * ------------------------------------------------------------------------
+ * Shared-memory usage statistics & metrics:
+ *
+ * Set of usage-stats fields copied from shmem_info{} struct, so that we
+ * can print these after shared segment has been destroyed.
+ * ------------------------------------------------------------------------
+ */
+typedef struct shminfo_usage_stats {
```

Comment:
Consolidated all metrics, usage-stats into this common struct.

It will be update in-place nested in the shmem control block.

And will also be used to return metrics when shared memory is being dismantled.

Works much better this way!!